scheduler/core/components/optimizer/base.py

```diff
@@ -2,9 +2,9 @@
 # For license information see LICENSE or https://opensource.org/licenses/BSD-3-Clause
 
 from abc import ABC, abstractmethod
-from datetime import datetime
-from typing import Mapping, List, Optional, Tuple
-
+from dataclasses import dataclass
+from datetime import timedelta
+from typing import Mapping, List, Optional, Union
```

Comment:
Just a reminder: we can omit `Union` from now on and just use `|` as in `int | str` for example.